src/common/summarize-profile.js

```diff
@@ -110,17 +97,43 @@ export function summarizeProfile(profile: Profile) {
  */
 function functionNameCategorizer() {
   const cache = new Map();
+
+  // Compile set of categories into a single regular expression that produces
+  // the pattern (whether it's a prefix, stem, substring, or exact match).
+
+  const patternToCategory = new Map();
+
+  const matchPatterns = {
+    prefix: [],
+    stem: [],
+    substring: [],
+    exact: []
+  };
+
+  for (const [matchType, pattern, category] of categories) {
+    patternToCategory.set(pattern, category);
+    let pat = pattern.replace(/\W/g, '\\$&');
+    if (matchType == 'stem')
+      pat += "\\b(?=\\()?";
```

Comment:
Yes, `stem` is pretty much for the full function name. Honestly, all of this is just a pile of hacks I came up with based on the function names observed in practice.

You're right, the optional lookahead is a complex noop. A simple \b is better.

If I put this in the regexp constructor thing, then I'd have to do something like

                           + '|\\b(?:' + [s + "\\b" for s in matchPatterns.stem].join('|') + ')'

and that would mess with the line wrapping and visual layout. :-)